packages/rocketchat-livechat/.app/imports/client/visitor.js

```diff
@@ -18,6 +18,23 @@ export default {
 		this.token.set(localStorage.getItem('visitorToken'));
 	},
 
+	reset() {
+		this.id.set(null);
+		this.token.set(null);
+		this.room.set(null);
+		this.data.set(null);
+		this.roomToSubscribe.set(null);
+		this.roomSubscribed = null;
+
+		Livechat.room = null;
+		Livechat.department = null;
+		Livechat.agent = null;
+		Livechat.guestName = null;
+		Livechat.guestEmail = null;
+		
+		msgStream.unsubscribe('room-messages');		
```

Comment:
this does not have the desired effect. on my tests after changing the token using `setGuestToken`, I'm still receiving messages from the old room/token and being notified as well.